slapp/transforms/transform_pipeline.py

```diff
@@ -179,30 +195,114 @@ def set_webm_parallelization(self, data, **kwargs):
+
+    # get all ROI ids from this segmentation run
+    query_string = ("SELECT id FROM rois WHERE segmentation_run_id="
+                    f"{segmentation_run_id}")
+    entries = db_conn.query(query_string)
+    roi_ids = [i['id'] for i in entries]
+
+    rois = [ROI.roi_from_query(roi_id, db_conn) for roi_id in roi_ids]
+
+    # load and downsample the source video
+    query_string = ("SELECT * FROM segmentation_runs "
+                    f"WHERE id={segmentation_run_id}")
+    seg_query = db_conn.query(query_string)[0]
+
+    return (rois, Path(seg_query['source_video_path']))
+
+
+class ProdSegmentationRunManifestSchema(mm.Schema):
+    experiment_id = mm.fields.Int(required=True)
+    binarized_rois_path = argschema.fields.InputFile(required=True)
+    # TODO: Consider using H5InputFile field from ophys_etl_pipelines
+    traces_h5_path = mm.fields.Str(required=True)
+    # TODO: Consider using H5InputFile field from ophys_etl_pipelines
+    movie_path = argschema.fields.Str(required=True)
+    local_to_global_roi_id_map = mm.fields.Dict(required=True,
+                                                keys=mm.fields.Int(),
+                                                values=mm.fields.Int())
+
+    @mm.post_load
+    def load_rois(self, data, **kwargs) -> dict:
+        with open(data['binarized_rois_path'], 'r') as f:
+            data['binarized_rois'] = json.load(f)
+        return data
+
+    @mm.post_load
+    def get_movie_frame_shape(self, data, **kwargs) -> dict:
+        with h5py.File(data['movie_path'], 'r') as h5f:
+            data['movie_frame_shape'] = h5f['data'].shape[1:]
+        return data
+
+
+def xform_from_prod_manifest(prod_manifest_path: str
+                             ) -> Tuple[List[ROI], Path]:
+    with open(prod_manifest_path, 'r') as f:
+        prod_manifest = json.load(f)
+    prod_manifest = ProdSegmentationRunManifestSchema().load(prod_manifest)
+    id_map = prod_manifest['local_to_global_roi_id_map']
+
+    rois = []
+    for roi in prod_manifest['binarized_rois']:
+        if roi['id'] not in id_map:
+            # only make manifests for listed ROIs
+            continue
+        # ROI mask_matrix is in 'postage stamp' form so row and col
+        # locations of data will need to have offset applied
+        roi_stamp = coo_matrix(roi['mask_matrix'])
+
+        converted_roi_id = id_map[roi['id']]
+
+        # TODO: Could possibly be replaced with _munge_traces()
+        # from ophys_etl_pipelines classification.py
+        with h5py.File(prod_manifest['traces_h5_path'], 'r') as h5f:
+            trace_data = h5f['data']
+            traces_id_order = h5f['roi_names'][:].astype(int)
+            traces_id_mapping = {v: i for i, v in enumerate(traces_id_order)}
+            assert roi['id'] == traces_id_order[traces_id_mapping[roi['id']]]
+            roi_trace = trace_data[traces_id_mapping[roi['id']]]
+
+        # TODO: check if offset applied may be off by 1
```

Comment:
You should be able to just add this assertion in the test you already have, no? Seems important.

Reply:
yep, will do. looks like this is left over from going fast

Reply:
done

Reply:
Can you point me to the line? I'm not seeing what I would expect in the test.

Reply:
sorry, here it is:
https://github.com/AllenInstitute/segmentation-labeling-app/blob/94beca88bf24a74a47f2728bd42eef4f66cce2bb/tests/transforms/test_transform_pipeline.py#L157-L163

Reply:
That's showing the trace, but I'm talking about the ROI rows and columns. I thought that was what the "offset may be off by 1" was referencing.

Reply:
ah, sorry, I was thrown off by "this assertion", I thought you were referring to the assertion in the code, not the `TODO`.
What he's doing here is correct, I think he checked it as well. I can add a test or something.

Reply:
@kschelonka see the final commit, I segmented this functionality out and tested it.